packages/nuxt/src/pages/module.ts

```diff
@@ -67,6 +73,70 @@ export default defineNuxtModule({
       return
     }
 
+    if (useExperimentalTypedPages) {
+      const declarationFile = './types/typed-router.d.ts'
+
+      const options: TypedRouterOptions = {
+        routesFolder: [],
+        dts: resolve(nuxt.options.buildDir, declarationFile),
+        logs: nuxt.options.debug,
+        async beforeWriteFiles (rootPage) {
+          rootPage.children.forEach(child => child.delete())
+          const pages = await resolvePagesRoutes()
+          await nuxt.callHook('pages:extend', pages)
+          function addPage (parent: EditableTreeNode, page: NuxtPage) {
+            // @ts-expect-error TODO: either fix types upstream or figure out another
+            // way to add a route without a file, which must be possible
+            const route = parent.insert(page.path, page.file)
+            if (page.meta) {
+              route.addToMeta(page.meta)
+            }
+            if (page.alias) {
+              route.addAlias(page.alias)
+            }
+            if (page.name) {
+              route.name = page.name
+            }
+            // TODO: implement redirect support
+            // if (page.redirect) {}
+            if (page.children) {
+              page.children.forEach(child => addPage(route, child))
+            }
+          }
+
+          for (const page of pages) {
+            addPage(rootPage, page)
+          }
+        }
+      }
+
+      nuxt.hook('prepare:types', ({ references }) => {
+        // This file will be generated by unplugin-vue-router
+        references.push({ path: declarationFile })
+      })
+
+      const context = createRoutesContext(resolveOptions(options))
```

Comment:
I need to take a look, I think the options should be resolved internally anyway 🤔